lib/miq_automation_engine/engine/miq_ae_service_model_base.rb

```diff
@@ -94,9 +96,9 @@ def self.expose(*args)
           return options[:override_return] if options.key?(:override_return)
           reflection = @object.class.reflection_with_virtual(method)
           if reflection && reflection.collection?
-            wrap_results(ret.to_a)
+            options[:association] ? wrap_results(self.class.filter_objects(ret.to_a)) : wrap_results(ret.to_a)
           else
-            wrap_results(ret)
+            options[:association] ? wrap_results(self.class.filter_objects(ret)) : wrap_results(ret)
```

Comment:
rbac now supports associations.
If possible please do not `.to_a` it just to pass to rbac. It will have to query the database and pull back the same rows again (less the rbac filter)